Client/Frontend/Home/RemoteTabsPanel.swift

```diff
@@ -82,29 +82,43 @@ class RemoteTabsPanel: UITableViewController, HomePanel {
             return
         }
 
-        // Otherwise, fetch the tabs cloud
-        self.profile.getClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
-            self.refreshControl?.endRefreshing()
+        // Otherwise, fetch the tabs cloud if its been more than 1 minute since last sync
+        let lastSyncTime = profile.prefs.timestampForKey("lastRemoteTabSyncTime")
+        if (lastSyncTime == nil) || NSDate.now() - lastSyncTime! > OneMinuteInMilliseconds {
+            self.profile.getClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
+                if let success = result.successValue {
+                    self.profile.prefs.setTimestamp(NSDate.now(), forKey: "lastRemoteTabSyncTime")
+                }
+                self.updateUIWithClientAndTabs(result)
+            }
+        } else {
+            self.profile.getCachedClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
```

Comment:
Maybe even consider

```
if (recent) {
  if (load) {
    return
  }
  if (!fetch) {
    load
  }
}
```

the idea being that it's better to fetch if we fail to load, and it's better to show stale than to fail to fetch. Maybe.